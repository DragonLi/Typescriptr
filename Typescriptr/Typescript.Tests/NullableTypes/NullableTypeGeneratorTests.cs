using System;
using Assent;
using Typescriptr;
using Xunit;

namespace Typescript.Tests.NullableTypes
{
    public class NullableTypeGeneratorTests
    {
        class TypeWithNullable
        {
            public int? NullableInt { get; set; }
            public Guid? NullableGuid { get; set; }
        }

        [Fact]
        public void Generator_TypeWithNullable_GeneratesSuccessfully()
        {
            var generator = TypeScriptGenerator.CreateDefault();
            var generated = generator.Generate(new[] {typeof(TypeWithNullable)});

            this.Assent(generated.Types);
        }
        
        class TypeWithNestedNullable
        {
            public class NestedType
            {
                public int? NullableInt { get; set; }
                public DateTime NullableDateTime { get; set; }
            }

            public NestedType NestedThing { get; set; }
        }

        [Fact]
        public void Generator_TypeWithNestedNullable_GeneratesSuccessfully()
        {
            var generator = TypeScriptGenerator.CreateDefault();
            var generated = generator.Generate(new[] {typeof(TypeWithNestedNullable)});

            this.Assent(generated.Types);
        }
    }
}