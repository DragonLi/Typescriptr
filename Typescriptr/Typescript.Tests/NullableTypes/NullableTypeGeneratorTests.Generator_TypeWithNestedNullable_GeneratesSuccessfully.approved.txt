declare namespace Api {
  interface TypeWithNestedNullable {
    nestedThing: NestedType;
  }
  interface NestedType {
    nullableInt?: number;
    nullableDateTime: string;
  }
}
